Core/Core/ViewModifiers/Badge.swift

```diff
@@ -0,0 +1,46 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+struct BadgeModifier<Content: View>: View {
+    let value: Int
+    let content: Content
+
+    var body: some View {
+        content.overlay(GeometryReader { geometry in
+            Text(NumberFormatter.localizedString(from: NSNumber(value: value), number: .none))
+                .font(.system(size: 10, weight: .semibold)).foregroundColor(.white)
+                .frame(minWidth: 14, maxHeight: 14)
+                .background(RoundedRectangle(cornerRadius: 7).stroke(Color.backgroundLightest, lineWidth: 2))
```

Comment:
there are mask & clip modifiers that could really cut out around the badge in the original view, but I didn't bother. It should be easy to add later if it looks much nicer in practice.